### DISCUSSION TITLE
Default behavior of bindings

### DISCUSSION THREAD

**Discussion by trashhacker** [AvaloniaUI/Avalonia#17179] [Q&A] [answered]
what is the default behavior of bindings with mode twoway. 
Initial conditions?

1. is the value to VM set from Control?
2. is the value to Control set from VM?

or the better question is why are my bindings get reseted ?

**Comment by IanRawley** [comment]
Control values are initially set to match VM values.  But you do have to be careful with SelectingItemControls such as ListBox or ComboBox.  Make sure you bind the ItemsSource before the SelectedItem, or the control will try to select an Item that isn't in its collection and refuse it, sending null back to the VM.

**Reply by trashhacker** [reply, marked as answer]
thanks a lot
